### ISSUE TITLE
Enable custom filters for Envoy

### ISSUE THREAD

**Issue by jamsajones** [aws/aws-app-mesh-roadmap#41]
Customers should be able to build their own filters into Envoy and we should allow config of those filters.

**Comment by donovano** [comment]
This question was answered somewhere else thanks. Being able to customize (i.e. add a custom filter) the AWS SigV4 Envoy and using it in your App Mesh will be a powerful feature. I believe the configuration for the custom filter would be done via the "static bootstrap config" which is fine - maybe one can add dynamic config in the future too. Looking forward to getting access to the SigV4 Envoy source and trying this out. Thanks.

**Comment by mattbailey** [comment]
Yes, I would love to see this too, envoy has a very powerful feature set that I'd like to use beyond app mesh integration. Similarly, I want gRPC-WEB, gRPC-JSON, and some of the authorization features.

**Comment by stanfan1992** [comment]
gRPC-JSON and gRPC-WEB and auth feature is definitely a good choice. 
Some use case and example or walkthrough will be even better.

**Comment by martijnalexion** [comment]
Our use-case is to use AppMesh with RDS. 
So we can access the databases on a different port than the default port.

**Comment by buffdaddy570** [comment]
I'd like to use the OPA Envoy plugin for external authorization of requests as talked about [here on the OPA website](https://www.openpolicyagent.org/docs/latest/envoy-introduction/). I believe it's related to this feature request.

**Comment by Dogacel** [comment]
I am extremely dissatisfied to see this feature has been lacking over the last 6 years. I hope AWS will make the right decision to make their image customizable by providing a config to merge.

Our current system runs two envoy proxies just for allowing gRPC transcoding. Can't wait to simplify our system and decrease our load + latency by shrinking it down to a single one.
